api/controllers/FlairController.js

```diff
@@ -154,7 +154,7 @@ module.exports = {
 
       for (var i = 0; i < flairs.fcs.length; i++) {
         let fc = flairs.fcs[i];
-        if (!Flairs.validFC(fc) && _.contains(req.user.loggedFriendCodes, fc)) {
+        if (!Flairs.validFC(fc) && flairs.fcs[i] !== req.user.loggedFriendCodes[i]) {
```

Comment:
Ok, I get the fact that it should be !, that would have been a mistake on my part when I was changing it.

However, I didn't realise we only wanted to check the most recent one.